src/main/java/com/browserstack/automate/ci/jenkins/pipeline/AppUploadStepExecution.java

```diff
@@ -42,9 +42,8 @@ protected Void run() throws Exception {
             return null;
         }
 
-        EnvVars overrides = run.getEnvironment(taskListener);
-        overrides.put(BrowserStackEnvVars.BROWSERSTACK_APP_ID, appId);
-        HashMap<String, String> overridesMap = new HashMap<>(overrides);
+        HashMap<String, String> overridesMap = new HashMap<String, String>();
+        overridesMap.put(BrowserStackEnvVars.BROWSERSTACK_APP_ID, appId);
```

Comment:
Wouldn't this remove any other values already present in run.getEnvironment(taskListener) or are those not required?

Reply:
^ yep. They will in case of Pipeline execution.

The thing is, this was the code earlier. I had modified it to include the existing values present in `run.getEnvironment(taskListener)`. But this resulted in a weird behaviour in which the user was trying to override the env vars inside a block and they weren't getting overridden (in `BrowserStackPipelineStepExecution`). 

But `AppUploadStepExecution` doesn't require any existing vars... thus changing back the code the way it was earlier.

Reply:
ok